applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.cpp

```diff
@@ -0,0 +1,181 @@
+    for(unsigned int i = 0; i < NumGPoints; ++i)
+        (JContainer[i]).resize(TDim,LocalDim,false);
+    Geom.Jacobian( JContainer, mThisIntegrationMethod );
+    
+    //Condition variables
+    array_1d<double,TNumNodes> NormalFluxVector;
+    for(unsigned int i = 0; i < TNumNodes; ++i)
+    {
+        NormalFluxVector[i] = Geom[i].FastGetSolutionStepValue(NORMAL_HEAT_FLUX);
+    }
+    NormalFluxVariables Variables;
+    
+    //Loop over integration points
+    for(unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+        //Compute normal flux 
+        Variables.NormalFlux = 0.0;
+        for (unsigned int i = 0; i < TNumNodes; ++i) {
+            Variables.NormalFlux += NContainer(GPoint,i) * NormalFluxVector[i];
+        }
+        
+        //Obtain Np
+        noalias(Variables.Np) = row(NContainer,GPoint);
+                
+        //Compute weighting coefficient for integration
+        this->CalculateIntegrationCoefficient(Variables.IntegrationCoefficient,
+                                              JContainer[GPoint],
+                                              IntegrationPoints[GPoint].Weight() );
+                
+        //Contributions to the right hand side
+        this->CalculateAndAddRHS(rRightHandSideVector, Variables);
+    }
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim,TNumNodes>::CalculateAndAddRHS(
+    VectorType& rRightHandSideVector,
+    NormalFluxVariables& rVariables )
+{
+    noalias(rVariables.TVector) = rVariables.NormalFlux * rVariables.Np * rVariables.IntegrationCoefficient;
+
+    GeoElementUtilities::
+        AssemblePBlockVector<0, TNumNodes>(rRightHandSideVector,
+                                                rVariables.TVector);
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<2, 2>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    const double dx_dxi = Jacobian(0, 0);
+    const double dy_dxi = Jacobian(1, 0);
+
+    const double ds = std::sqrt(dx_dxi * dx_dxi + dy_dxi * dy_dxi);
+
+    rIntegrationCoefficient = ds * Weight;
+}
+
+// ============================================================================================
+// ============================================================================================
+template< >
+void TNormalFluxCondition<2, 3>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    const double dx_dxi = Jacobian(0, 0);
```

Comment:
I think you can call the other function instead of repeating the code, i.e.:

`this->CalculateIntegrationCoefficient<2,2>(rIntegrationCoefficient, Jacobian, Weight);`

Reply:
I did not succeed to call the other function, as I get syntax errors.
However, it is possible to define another function, and call the new generated function by both of thos function, namely by <2,2> and <2,3>.
Because the code becomes less clear by introducing a new function, I left the implementation as it it.